go/adbc/driver/databricks/command.go

```diff
@@ -0,0 +1,384 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package databricks
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+)
+
+const (
+	CmdOptionIntByteLimit  = "adbc.databricks.byte_limit"
+	CmdOptionIntRowLimit   = "adbc.databricks.row_limit"
+	CmdOptionBoolTruncated = "adbc.databricks.truncated" // read-only
+)
+
+type command struct {
+	alloc memory.Allocator
+	conn  *connectionImpl
+
+	req *compute.Command
+
+	commandId string
+}
+
+func NewCommand(conn *connectionImpl) (adbc.Statement, error) {
+	cluster, err := conn.client.Clusters.Get(context.Background(), compute.GetClusterRequest{
+		ClusterId: conn.client.Config.ClusterID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if cluster.State == compute.StateTerminated || cluster.State == compute.StateTerminating {
+		wait, err := conn.client.Clusters.Start(context.Background(), compute.StartCluster{
+			ClusterId: conn.client.Config.ClusterID,
+		})
```

Comment:
Are you sure it's a good idea to do this here? I think we should return an error with `adbc.StatusCancelled` instead.

A good place to put the cluster starting command is in the new connection logic. So we can say connection time includes start up time, but command execution either finds a started cluster or doesn't work because we need to connect again (i.e. restart the cluster).

Imagine client code firing 1000s of stmts on multiple connections, all trying to start a terminating cluster.